cyral/data_source_cyral_integration_idp.go

```diff
@@ -0,0 +1,167 @@
+			"display_name": {
+				Description: "Filter results by the name of an existing IdP integration.",
+				Optional:    true,
+				Type:        schema.TypeString,
+			},
+			"type": {
+				Description: "Filter results by the IdP integration type.",
+				Optional:    true,
+				Type:        schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{
+					"aad",
+					"adfs",
+					"forgerock",
+					"gsuite",
+					"okta",
+					"pingone",
+				}, false),
+			},
+			"idp_list": {
+				Type: schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"display_name": {
+							Description: "Display name used in the Cyral control plane.",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
+						"alias": {
+							Description: "Internal alias (ID) for this integration.",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
+						"single_sign_on_service_url": {
+							Description: "Single sign on service URL.",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
+						"enabled": {
+							Description: "True if the IdP integration is enabled.",
+							Type:        schema.TypeBool,
+							Computed:    true,
+						},
+					},
+				},
+				Computed:    true,
+				Description: "Set of existing IdP integrations for the given filter criteria.",
+			},
+		},
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+	}
+}
+
+func dataSourceIntegrationIdPRead(
+	ctx context.Context,
+	d *schema.ResourceData,
+	m interface{},
+) diag.Diagnostics {
+	log.Printf("[DEBUG] Init dataSourceIntegrationIdPRead")
+	c := m.(*client.Client)
+
+	idpTypeFilter := d.Get("type").(string)
+
+	var url = fmt.Sprintf("https://%s/v1/integrations/saml", c.ControlPlane)
+	if idpTypeFilter != "" {
+		url = fmt.Sprintf("https://%s/v1/integrations/saml?identityProvider=%s",
+			c.ControlPlane, idpTypeFilter)
+	}
+
+	body, err := c.DoRequest(url, http.MethodGet, nil)
+	if err != nil {
+		return createError("Unable to execute request to read idp integrations", err.Error())
+	}
+
+	var idpIntegrations = IdPIntegrations{}
+	if err := json.Unmarshal(body, &idpIntegrations); err != nil {
+		return createError("Unable to unmarshal idp integrations", err.Error())
+	}
+	log.Printf("[DEBUG] Response body (unmarshalled): %#v", idpIntegrations)
+
+	var idpList []interface{}
+	displayNameFilter := d.Get("display_name").(string)
+
+	log.Printf("[DEBUG] display_name: %s", displayNameFilter)
+	log.Printf("[DEBUG] type: %s", idpTypeFilter)
+	if idpIntegrations.Connections != nil {
+		for _, connection := range idpIntegrations.Connections.Connections {
+			log.Printf("[DEBUG] Connection: %#v", connection)
+			if connection != nil {
+				// Skip in case filters are non-empty but
+				if (displayNameFilter != "" && displayNameFilter != connection.DisplayName) ||
+					(idpTypeFilter != "" && !strings.HasPrefix(connection.Alias, idpTypeFilter)) {
```

Comment:
Since the API is already filtering by type, this line is redundant
```suggestion
					(idpTypeFilter != "" && !strings.HasPrefix(connection.Alias, idpTypeFilter)) {
```